src/OrleansSQLUtils/CreateOrleansTables_MySql.sql

```diff
@@ -441,77 +408,26 @@ BEGIN
 			)
 			VALUES
 			(
-				_serviceId,
-				_grainId,
-				_reminderName,
-				_startTime,
-				_period,
-				_grainIdConsistentHash
-			);
-		ELSE
-			UPDATE OrleansRemindersTable
-			SET				
-				StartTime             = _startTime,
-				Period                = _period,
-				GrainIdConsistentHash = _grainIdConsistentHash,
-				ETag                  = _newEtag
-			WHERE
-				ServiceId = _serviceId AND _serviceId IS NOT NULL
-				AND GrainId = _grainId AND _grainId IS NOT NULL
-				AND ReminderName = _reminderName AND _reminderName IS NOT NULL;
-		END IF;
-		COMMIT;
-		SELECT CONVERT(_newEtag, BINARY) AS ETag;
-END$$
-
-DELIMITER ;
+				@serviceId,
+				@grainId,
+				@reminderName,
+				@startTime,
+				@period,
+				@grainIdConsistentHash
+			)
+		ON DUPLICATE KEY UPDATE 
```

Comment:
I wonder, here and in the other updates on duplicate key if a conditional would be needed, as you speculate at https://github.com/dotnet/orleans/issues/1169 so that not all rows are updated?